src/domain/collection/curatedEvents/__tests__/CuratedEvents.test.tsx

```diff
@@ -11,106 +11,96 @@ import {
-          };
-        })
-      )
-    ),
-  ])(eventsCount);
+afterAll(() => {
+  clear();
+});
 
-  const mocks = chunkedEvents.map((eventList) => {
-    return getMocks(
-      eventList.data as EventFieldsFragment[],
-      eventList.data.map((event) => event.id)
-    )[0];
-  });
+afterEach(() => {
+  jest.restoreAllMocks();
+});
 
-  const collection = fakeCollection({
-    curatedEvents,
-  }) as CollectionFieldsFragment;
+[true, false].forEach((EVENTS_HELSINKI_2) => {
+  test(`should show all events (EVENT_HELSINKI_2 feature is ${
+    EVENTS_HELSINKI_2 ? 'on' : 'off'
+  })`, async () => {
+    setFeatureFlags({ EVENTS_HELSINKI_2 });
+    const events = fakeEvents(
+      eventNames.length,
+      eventNames.map((event, index) => ({
+        id: eventIds[index],
+        name: { fi: event },
+      }))
+    );
+    const courses = fakeEvents(
+      courseNames.length,
+      courseNames.map((event, index) => ({
+        id: courseIds[index],
+        name: { fi: event },
+      }))
+    );
+    const eventsData = events.data as EventFieldsFragment[];
+    const coursesData = courses.data as EventFieldsFragment[];
+    const eventMocks = getMocks(eventsData, eventIds, 'event');
+    const courseMocks = getMocks(coursesData, courseIds, 'course');
 
-  return { mocks, collection, eventNames };
-};
+    render(<CuratedEvents collection={collection} />, {
+      mocks: [...eventMocks, ...courseMocks],
+    });
 
-const waitForRequestToComplete = async () => {
-  await waitFor(() => {
-    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
-  });
-};
+    await waitFor(() => {
+      expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+    });
 
-afterAll(() => {
-  clear();
-});
+    // Test that both lists are
 
-test('should show all events', async () => {
-  const events = fakeEvents(
-    eventNames.length,
-    eventNames.map((event, index) => ({
-      id: eventIds[index],
-      name: { fi: event },
-    }))
-  );
-  const eventsData = events.data as EventFieldsFragment[];
-  const mocks = getMocks(eventsData);
+    const eventsList = within(screen.getByTestId(eventsListTestId));
+    eventNames.forEach((eventName) => {
+      expect(eventsList.getByText(eventName)).toBeInTheDocument();
+    });
 
-  render(<CuratedEventList collection={collection} />, {
-    mocks,
-  });
+    await waitFor(() => {
+      expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+    });
 
-  await waitFor(() => {
-    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
-  });
-  eventNames.forEach((eventName) => {
-    expect(screen.getByText(eventName)).toBeInTheDocument();
+    if (isFeatureEnabled('EVENTS_HELSINKI_2')) {
+      const courseList = within(screen.getByTestId(coursesListTestId));
+      courseNames.forEach((courseName) => {
+        expect(courseList.getByText(courseName)).toBeInTheDocument();
+      });
+    }
```

Comment:
`else` check is missing. Without else there is no reason to have the loop `[true, false].forEach((EVENTS_HELSINKI_2)` because when it's `false` it does not expect anything more in addition to `true`.
Perhaps either remove the loop or add else check:
```
if (isFeatureEnabled('EVENTS_HELSINKI_2')) {
...
} else {
    expect(screen.queryByTestId(coursesListTestId)).not.toBeInTheDocument();
}
```

Reply:
true, good to add that check also

Reply:
Well it did test that it works without the flag :) 

But fixed now